com.unity.shadergraph/Editor/Drawing/PreviewManager.cs

```diff
@@ -486,10 +488,29 @@ public void RenderPreviews(bool requestShaders = true)
                 CollectPreviewProperties(m_NodesPropertyChanged, perMaterialPreviewProperties);
                 m_NodesPropertyChanged.Clear();
 
-                // timed nodes change every frame, so must be drawn
+                // timed nodes are animated, so they should be updated regularly
                 // (m_TimedPreviews has been pre-propagated downstream)
                 // HOWEVER they do not need to collect properties. (the only property changing is time..)
-                m_PreviewsToDraw.UnionWith(m_TimedPreviews);
+
+                // get current window FPS, clamp to what we consider a valid range
```

Comment:
Seems cleaner if this is all moved to a `bool NeedsTimeUpdate(editorWindow)` function - this function is already pretty long.

Reply:
Moved it!